crates/matrix-sdk-ui/src/timeline/tests/event_filter.rs

```diff
@@ -200,3 +203,120 @@ async fn hide_failed_to_parse() {
 
     assert_eq!(timeline.inner.items().await.len(), 0);
 }
+
+#[async_test]
+async fn event_type_filter_include_only_room_names() {
+    // Only return room name events
+    let event_filter = TimelineEventTypeFilter::Include(vec![TimelineEventType::RoomName]);
+
+    let timeline = TestTimeline::new().with_settings(TimelineInnerSettings {
+        event_filter: Arc::new(move |event, _| event_filter.filter(event)),
+        ..Default::default()
+    });
+
+    // Add a non-encrypted message event
+    timeline
+        .handle_live_message_event(&ALICE, RoomMessageEventContent::text_plain("The first message"))
+        .await;
+    // And a couple of room name events
+    timeline
+        .handle_live_state_event(
+            &ALICE,
+            RoomNameEventContent::new("A new room name".to_string()),
+            None,
+        )
+        .await;
+    timeline
+        .handle_live_state_event(
+            &ALICE,
+            RoomNameEventContent::new("A new room name (again)".to_string()),
+            None,
+        )
+        .await;
+
+    // The timeline should contain only the room name events
+    let event_items: Vec<Arc<TimelineItem>> = timeline.get_event_items().await;
+    let text_message_items: Vec<Arc<TimelineItem>> =
+        event_items.clone().into_iter().filter(is_text_message_item).collect();
+    let room_name_items: Vec<Arc<TimelineItem>> =
+        event_items.clone().into_iter().filter(is_room_name_item).collect();
```

Comment:
We could simplify this a bit with something like:

```
let num_text_messages = event_items.iter().filter(is_text_message_item).count();
// ditto for `num_room_name_items`
```

and check against those numbers directly below